File: src/main/java/com/matag/game/player/PlayerUpdateEvent.java
package com.matag.game.player;

import com.matag.game.cardinstance.CardInstance;
import lombok.Builder;
import lombok.Data;

import java.util.List;

@Data
@Builder
public class PlayerUpdateEvent {
  String name;
  int life;
  int librarySize;
  List<CardInstance> visibleLibrary;
  List<CardInstance> hand;
  List<CardInstance> battlefield;
  List<CardInstance> graveyard;
}

File: src/main/kotlin/com/matag/game/player/PlayerUpdateEvent.kt
package com.matag.game.player

import com.matag.game.cardinstance.CardInstance

data class PlayerUpdateEvent(
    var name: String? = null,
    var life: Int = 0,
    var librarySize: Int = 0,
    var visibleLibrary: List<CardInstance> = listOf(),
    var hand: List<CardInstance> = listOf(),
    var battlefield: List<CardInstance> = listOf(),
    var graveyard: List<CardInstance> = listOf()
)
